test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -2456,7 +2456,7 @@ public synchronized void assertAfterTest() throws Exception {
                 try {
                     env.shardLock(id, "InternalTestCluster assert after test", TimeUnit.SECONDS.toMillis(5)).close();
                 } catch (ShardLockObtainFailedException ex) {
-                    fail("Shard " + id + " is still locked after 5 sec waiting");
+                    fail("Shard " + id + " is still locked after 5 sec waiting: " + ex.getMessage());
```

Comment:
Maybe use this instead:
```suggestion
throw new AssertionError("Shard " + id + " is still locked after 5 sec waiting", ex);
```

unless it is too verbose?